src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -339,14 +340,21 @@ private void doRetrieve(SCMHeadObserver observer, TaskListener listener, GHRepos
                 return 0; // TODO
             }
             @Override public boolean exists(@Nonnull String path) throws IOException {
-                try {
-                    repo.getFileContent(path, ref);
-                    listener.getLogger().format("      %s exists in this %s%n", path, thing);
-                    return true;
-                } catch (FileNotFoundException x) {
-                    listener.getLogger().format("      %s does not exist in this %s%n", path, thing);
-                    return false;
+                List<GHContent> directoryContent = repo.getDirectoryContent("/", ref);
```

Comment:
@jglick I assumed that `SCRIPT_FILE` should be placed in the root. But the method description is clear:

```
/**
 * Checks if the path, relative to the head candidate root, exists or not. The results of this method should
 * be cached where possible but can involve a remote network call.
 *
 * @param path the path.
 * @return {@code true} iff the path exists (may be a file or a directory or a symlink or whatever).
 * @throws IOException if a remote network call failed and the result is therefore indeterminate.
 */
```

I'm going to review the current implementation.